### PR TITLE
!!!TASK: Disable and deprecate plugin and content prototype generators

### PR DESCRIPTION
**Pull request by mficzel**
The prototype generator classes are deprecated and removed from the nodeType configuration. This
helps to make the fusion-code more explict and understand the connection between nodeType and fusion prototype much simpler. The previously assigned base types were not recommended anymore and thus usually are overwritten anyways.

A very welcome effect is that the error message for missing node rendering prototypes will say just that and not any more that a template file is missing.

In addition this removes the documentPrototypeGenerator that was deprecated already for Neos 5.

Upgrade instructions:

To upgrade you have to make sure that content and plugin nodetypes have a matching fusion prototype that looks like the following examples:

```
prototype(Vendor.Site:ContentExample) < prototype(Neos.Fusion:Template) {
    templatePath = "resource://Vendor.Site/Private/Templates/NodeTypes/EContentExamplexample.html" 
    /* for all properties needed for rendering */
    title = ${q(node).property('title')}
}

prototype(Vendor.Site:PluginExample) < prototype(Neos.Neos:Plugin) {
        package = "Vendor.Site"
        subpackage = ""
        controller = "Example"
        action = "index"
}
```
Note: `Neos.Fusion:Template` is not considered best practice anymore in favor of fusion-afx.

### CODE REVIEW

File: Neos.NodeTypes/Configuration/NodeTypes.Content.Headline.yaml [hunk 1]
```diff
@@ -6,3 +6,6 @@
     label: i18n
     icon: 'icon-header'
     position: 100
+  options:
+    fusion:
+      prototypeGenerator: Neos\Neos\Domain\Service\DefaultContentPrototypeGenerator
```
Comment (daniellienert) [inline Neos.NodeTypes/Configuration/NodeTypes.Content.Headline.yaml]: @mficzel: what do you think about adding the the prototype code directly to fusion files? It feels strange to still use the deprecated class internally.

Next steps could be to rewrite the Fluid to Fusion. If thats a good idea, that could be good-first-issues

Comment (mficzel) [reply Neos.NodeTypes/Configuration/NodeTypes.Content.Headline.yaml]: @daniellienert I added the prototype code to the fusion of the subpackages we like and are maintaining actively. 

For Neos.NodeTypes itself i took the minimal needed effort as we are saying for a while that Neos.NodeTypes should not be used anymore. It could be done and probably would be god but i would see it in a follow up pr.

**Comment by mficzel** [comment]
@kdambekalns @daniellienert i removed the prototype generators from the Neos.NodeTypes NodeTypes aswell. I finally got the idea to dump and copy paste the generated code which made this easy.

**Comment by kitsunet** [comment]
Wonderful, isn't the example in the message here misleading though? For content NodeTypes you would want to use ContentComponent ?

**Comment by kitsunet** [comment]
Would merge after above question is cleared.

File: Neos.Neos/Configuration/Testing/NodeTypes.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 'Acme.Demo:ThreeColumn':
   superTypes:
-    'Neos.NodeTypes:Column': true
+    'Neos.Neos:Content': true
```
Comment (bwaidelich) [inline Neos.Neos/Configuration/Testing/NodeTypes.yaml]: Why this change?

Comment (mficzel) [reply Neos.Neos/Configuration/Testing/NodeTypes.yaml]: Because there is no Neos.Nodetypes:Column any more. Also the tests should not rely on NeosNodetypes at all as we would really like to get rid of this package.

Comment (mficzel) [reply Neos.Neos/Configuration/Testing/NodeTypes.yaml]: @bwaidelich is this ok for you, i would like to merge this as there are 3 approvals already.

Comment (bwaidelich) [reply Neos.Neos/Configuration/Testing/NodeTypes.yaml]: Yes sure! And thanks for the explanation!
